### PR TITLE
Development tracer NuRadio

### PR DESCRIPTION
**Pull request by boboeyen**
Necessary alteration for a RadioPropa ray tracer in NuRadio

### CODE REVIEW

**Comment by boboeyen** [comment]
The necessary clean up of the repository will follow in a other Pull request :)

File: src/IceModel.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+double IceModel_Simple::getValue(const Vector3d &position) const
+{
+	if (position.z <= _z_surface) {
+		return _ice.getValue(position);
+	} else {
+		return 1.;
+	}
+}
+double IceModel_Simple::getAverageValue(const Vector3d &position1, const Vector3d &position2) const
+{
```
Comment (iplaisier) [inline src/IceModel.cpp:44]: What is the idea of getting the average for 2 positions of an analytic function? For what is this used?

File: include/radiopropa/IceModel.h [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+class IceModel_Firn: public ScalarField
+{
+	protected:
+		double _z_surface, _z_firn;
+		ExponentialIndex _firn;
+		ExponentialIndex _ice;
+	public:
+		IceModel_Firn(  
+			double n_ice_firn,
```
Comment (iplaisier) [inline include/radiopropa/IceModel.h:45]: can you document here also the paper where the (firn)model is taken from? Like how is done for the gorham icemodel.

File: extra_modules_python/DiscontinuityLayer.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import radiopropa
```
Comment (iplaisier) [inline extra_modules_python/DiscontinuityLayer.py:1]: This file can be removed.

File: include/RadioPropa.h [hunk 4]
```diff
@@ -17,6 +17,7 @@
 #include "radiopropa/Referenced.h"
 #include "radiopropa/Source.h"
 #include "radiopropa/ScalarField.h"
+#include "radiopropa/IceModel.h"
```
Comment (iplaisier) [inline include/RadioPropa.h:20]: Is there a good reason for separating IceModel and Scalarfield?

**Review by iplaisier — APPROVED** [review]
Looks fine to me!

File: src/IceModel.cpp (hunk 1 above)
Comment (boboeyen) [reply src/IceModel.cpp:44]: It was implemented in medium.py of NuRadioMC, so I stream lined it

File: include/radiopropa/IceModel.h (hunk 2 above)
Comment (boboeyen) [reply include/radiopropa/IceModel.h:45]: why should this be documented? It is just to exponential stacked on each other but with no parameters, the parameters specific to the paper are implemented in NuRadioMC

File: extra_modules_python/DiscontinuityLayer.py (hunk 3 above)
Comment (boboeyen) [reply extra_modules_python/DiscontinuityLayer.py:1]: clean up is done in the next pull request ;)

File: include/RadioPropa.h (hunk 4 above)
Comment (boboeyen) [reply include/RadioPropa.h:20]: It did it for transparency

**Pull request merged**